### ISSUE TITLE
Use of RFtime data unclear

### ISSUE THREAD

**Issue by sdobbs** [JeffersonLab/HDGeant4#22]
One issue I've run into seems to come from the inclusion of RFtime information in the output of HDGeant4.  This data was not generated in HDGeant.

I believe that the values stored come from GlueXPhotonBeamGenerator::GenerateRFsync().  It's not clear to me what these values are supposed to represent.  Are they an overall shift for all detector signals with respect to absolute t=0?

Currently, DEventSourceHDDM interprets RFtime data in its HDDM format as data corresponding to a DRFTime object, which as it stands is incorrect and seems to break any reconstruction that matches back to the RF time.

**Comment by pmattjlab** [comment]
It's true that HDGeant didn't generate it.  So, in sim-recon the 
"thrown" rf time is created in DEventSourceHDDM.cc for MC, such that it 
agrees with the true time at the center of the target of the beam photon 
that started the event.

I dunno what is going in HDGeant4, but hopefully it's doing that.

  - Paul

**Comment by rjones30** [comment]
Hello Sean,

My purpose in inserting RFtime is to have a sync that identifies the start
of a 4.096ns machine period, prescaled by the same factor as the real data,
and with an arbitrary offset from the bucket that generated the actual
triggered interaction. I believe that is what it does. That is what the
machine RFtime does for us in the real data, if I am not mistaken.

-Richard

**Comment by mashephe** [comment]
I don't mean to hijack this thread, but I have a question about DEventRFBunch...

Is there ever more or less than one of these objects per event in real data?  If so, then what is the advice for picking the one that is most likely to be correct?  

Matt

**Comment by pmattjlab** [comment]
With the default tag, there is always exactly one: what the library thought matched the event the best.  However, different combos may make a different selection ("Combo" tag). 

 - Paul

----- Original Message -----

**Comment by sdobbs** [comment]
Getting back to the original issue:
From Paul's and RIchard's responses, it seems clear that HDGeant4 and sim-recon are doing two different things, and interpreting the RFtime data in HDDM as a calibrated/reconstructed DRFTime object is the wrong thing to do, since the DRFTime corresponds to "true time at the center of the target of the beam photon that started the event"

I can think of two ways forward:
- Make a new DRFMCTime object which is reconstructed to a DRFTime object in a similar way as data
- Not writing out RFTime data anymore and reverting to the old way of time determination.

The second option seems simpler to me, but it's worth having a discussion.

**Comment by rjones30** [comment]
Sean,

I am puzzled by the description you provide. In the real events, there is a
prescaled copy of the machine 250 MHz RF signal that is digitized in
several of the subsystems, including the tagger. That is NOT identified in
time with any trigger, let alone with any guess of which bucket caused the
trigger. That can happen anywhere within the TDC digitization frame,
typically once per event. AFAIK, that thing is known as the machine RF.
What is that called in the jana object world? Whatever that is, that is
what the tag I called "RFtime" in the hddm_s mc record is supposed to
represent. From what I can tell reading through your discussion, that
signal is not in view. Is it?

-Richard Jones

**Comment by sdobbs** [comment]
Richard,

Yes, what you are describing is essentially equivalent to the DRFTDCDigiTime object (albeit measured in physical units).  However, currently the RFtime record is interpreted as a different type of object, so we would need to change how it's reconstructed for the times you are writing out to be properly interpreted.

**Comment by sdobbs** [comment]
And we'll need to add some smearing in mcsmear...

**Comment by sdobbs** [comment]
For what it's worth, I think I see an easier way to implement accommodate this than the procedure I outlined before...

**Comment by rjones30** [comment]
Sean,

How about if I drop writing out the RFtime tag? It is redundant information
in MC anyway, because the only use it has would be to tell us how to shift
all of the times in the event by something in the range [-2, +2] ns so that
there is a machine clock edge at t=0. But that is already true in my
convention for the MC event clock, so in fact this is redundant
information. The only reason I added it was because I saw the record in the
hddm_s event spec and thought it needed to be filled.

-Richard J.
